streamr-dev/streamr-platform
# [PLATFORM-1202] Community members list

Adds the community member join request listing.

![ezgif-2-280d64ef8eb0](https://user-images.githubusercontent.com/1064982/72518919-02e78f80-385f-11ea-89c2-142c86d938fb.gif)

UI notes:
- Shows full address on hover or if selected
- Toggling "select all" and "deselect all" restores values selected by the user, cancel drops all selections (is this how it should work @mattatgit ?)
- Toolbar has only action to approve since that's how it's in the design
- Paging not implemented (does it make sense if the list is long?)
- Current filtering options (Recent = Show all):

<img width="166" alt="Screen Shot 2020-01-16 at 13 10 04" src="https://user-images.githubusercontent.com/1064982/72520193-95892e00-3861-11ea-9839-60b28071d21d.png">

To test:
- Create and deploy a community, click "View community" from product tile
- To add members, either insert rows directly to database table `community_join_request` using product's beneficiary address as the community id and an ethereum id that is linked to a streamr account, or hook up a call somewhere to `addRequest`:

```
import useJoinRequests from '$mp/modules/communityProduct/hooks/useJoinRequests'
// ...
const { addRequest } = useJoinRequests()

addRequest(communityId, memberAddress)
```

Troubleshooting:
- Error ("illegal url" or something like that) during fetch -> update docker to latest e&e
- Error message from API: `"tester1@streamr.com does not have permission to manage community (id 0xasd...)"` -> Check `integration_key table` in database, search owner address from `integration_key.id_in_service` column, remove if not linked to current user (might be linked to anonymous user)
- Approving does not work atm, e&e fix pending
